astroplan/scheduling.py

```diff
@@ -350,22 +559,21 @@ def from_timespan(cls, center_time, duration, **kwargs):
         duration : `~astropy.units.Quantity` or `~astropy.time.TimeDelta`
             Duration of time-span to schedule
         """
-        start_time = center_time - duration/2.
-        end_time = center_time + duration/2.
+        start_time = center_time - duration / 2.
+        end_time = center_time + duration / 2.
         return cls(start_time, end_time, **kwargs)
 
     def _make_schedule(self, blocks):
-
         # Combine individual constraints with global constraints, and
         # retrieve priorities from each block to define scheduling order
         _all_times = []
         _block_priorities = np.zeros(len(blocks))
-        for i,b in enumerate(blocks):
+        for i, b in enumerate(blocks):
             if b.constraints is None:
                 b._all_constraints = self.constraints
             else:
                 b._all_constraints = self.constraints + b.constraints
-            b._duration_offsets = u.Quantity([0*u.second, b.duration/2, b.duration])
+            b._duration_offsets = u.Quantity([0 * u.second, b.duration / 2, b.duration])
```

Comment:
I could image extra keywords on `Constraints` indicating how fine of time spacing they need to be evaluated at, which could be handled by the schedulers. The issue is that it would not fit with the way that `Constraint`s are currently defined: to be evaluated at a single point, and that it would require more complex constraint evaluation by schedulers. If I can create an efficient way of allowing a `Slot` to calculate `Constraint` in itself, I could create exceptions for certain `Constraint`s so that they only needed evaluation at certain points.